polymath/polynomial.py

```
@@ -487,26 +604,65 @@ def eval(self, x, recursive=True):
+                const_values = self._values[..., 0]
+
             if recursive:
-                return Scalar(example=self)
+                # Convert derivatives from Polynomial to Scalar
+                derivs = {}
+                for key, deriv in self._derivs.items():
+                    # Derivative of a constant polynomial is also constant
+                    assert deriv.order == 0
+                    deriv_tail = deriv._drank * (slice(None),)
+                    if deriv_tail:
+                        deriv_const = deriv._values[(Ellipsis, 0) + deriv_tail]
```

Ruff (0.14.7)

Consider `(Ellipsis, 0, *deriv_tail)` instead of concatenation

Replace with `(Ellipsis, 0, *deriv_tail)`

(RUF005)